src/App.tsx

```diff
@@ -1,49 +1,85 @@
-import { SearchHeader } from "./components/header/SearchHeader";
-import { ProductsGrid } from "./components/productsGrid/ProductsGrid";
+import { SearchAndSortHeader } from "./components/header/SearchAndSortHeader";
+import { ProductsGrid } from "./components/products/ProductsGrid";
 import { Sidebar } from "./components/sidebar/Sidebar";
 import { useState } from "react";
-import { useDebounce } from "./hooks/useDebounce";
+import { useDebounce } from "./hooks";
 
 function App() {
-  const [isSidebarOpen, setIsSidebarOpen] = useState(false);
-  const [searchTerm, setSearchTerm] = useState("");
-  const [sortValues, setSortValues] = useState({
+  const [state, setState] = useState({
+    isSidebarOpen: false,
+    searchTerm: "",
+    label: "Sort by",
     sortBy: "",
     sortOrder: "",
+    isSortModalOpen: false,
   });
-  const { sortBy, sortOrder } = sortValues;
-  const debouncedSearchTerm = useDebounce(searchTerm, 500);
-  const [isSortModalOpen, setIsSortModalOpen] = useState(false);
+
+  const { sortBy, sortOrder, label } = state;
+  const debouncedSearchTerm = useDebounce(state.searchTerm, 500);
 
   const handleOpenSortModal = () => {
-    setIsSortModalOpen((prev) => !prev);
+    setState((prevState) => ({
+      ...prevState,
+      isSortModalOpen: !prevState.isSortModalOpen,
+    }));
   };
 
   const handleOpenSidebar = () => {
-    setIsSidebarOpen((prev) => !prev);
+    setState((prevState) => ({
+      ...prevState,
+      isSidebarOpen: !prevState.isSidebarOpen,
+    }));
   };
 
   const handleChangeSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
     const { value } = event.target;
-    setSearchTerm(value);
+    setState((prevState) => ({ ...prevState, searchTerm: value }));
   };
 
   const handleChangeSort = (event: React.MouseEvent<HTMLButtonElement>) => {
     const { value, name } = event.currentTarget;
-    setSortValues({ sortBy: name, sortOrder: value });
-    setIsSortModalOpen(false);
+
+    if (name === sortBy) {
+      setState((prevState) => ({
+        ...prevState,
+        label: "Sort by",
+        sortBy: "",
+        sortOrder: "",
+      }));
+    } else {
+      let label = "";
+      if (name === "favorites") {
+        label = "Favorites";
+      } else if (name === "price") {
+        label = value === "asc" ? "Price: Low to High" : "Price: High to Low";
+      } else if (name === "title") {
+        label = value === "asc" ? "Title: A to Z" : "Title: Z to A";
+      }
+
+      setState((prevState) => ({
+        ...prevState,
+        label,
+        sortBy: name,
+        sortOrder: value,
+      }));
+    }
+
+    setState((prevState) => ({ ...prevState, isSortModalOpen: false }));
```

Comment:
And you can refactor this function in order to improve readability, for example add an early return if the sortBy is the same and then use the helper function and  group all state updates in one call:

```
const handleChangeSort = (event: React.MouseEvent<HTMLButtonElement>) => {
  const { value, name } = event.currentTarget;

  // Return early if the sortBy is the same (reset sorting)
  if (name === sortBy) {
    setState((prevState) => ({
      ...prevState,
      label: "Sort by",
      sortBy: "",
      sortOrder: "",
      isSortModalOpen: false, // Close modal here instead of in another call
    }));
    return;
  }

  const label = getSortLabel(name as SortOptions, value as SortOrder);

  setState((prevState) => ({
    ...prevState,
    label,
    sortBy: name,
    sortOrder: value,
    isSortModalOpen: false, // Close modal in the same setState call
  }));
};
```